src/types/types.ts

```diff
@@ -3,8 +3,19 @@ import { SlackManifest } from "../manifest.ts";
 import { ManifestCustomTypeSchema } from "../types.ts";
 
 export type CustomTypeDefinition =
-  & { callback_id: string }
-  & TypedParameterDefinition;
+  | (
+    & {
+      /**
+       * @deprecated Use name instead
+       */
```

Comment:
This isn't doing what I would expect as it's not messaging the user as they use `DefineType`. We can come back to it in another PR.